client/state/themes/selectors/is-theme-purchased.js

```diff
@@ -15,5 +15,11 @@ import 'calypso/state/themes/init';
  */
 export function isThemePurchased( state, themeId, siteId ) {
 	const sitePurchases = getSitePurchases( state, siteId );
-	return some( sitePurchases, { productSlug: 'premium_theme', meta: themeId } );
+	return !! sitePurchases.find( ( purchase ) => {
```

Comment:
Minor, but FWIW you can use `Array.prototype.some` instead of `!! Array.prototype.find`:

```suggestion
	return sitePurchases.some( ( purchase ) => {
```